File: frontend/apps/android/app/src/main/java/app/pantopus/android/ui/screens/shared/wizard/blocks/ReviewSummaryBlock.kt
@file:Suppress("MatchingDeclarationName")

package app.pantopus.android.ui.screens.shared.wizard.blocks

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.unit.dp
import app.pantopus.android.ui.theme.PantopusColors
import app.pantopus.android.ui.theme.PantopusTextStyle
import app.pantopus.android.ui.theme.Radii
import app.pantopus.android.ui.theme.Spacing

/** A single label/value row in a [ReviewSummaryBlock]. */
data class ReviewSummaryRow(
    val label: String,
    val value: String,
)

/**
 * White card with a stack of label/value rows separated by hairlines.
 * Use on the wizard's review step.
 */
@Composable
fun ReviewSummaryBlock(
    rows: List<ReviewSummaryRow>,
    modifier: Modifier = Modifier,
) {
    Column(
        modifier =
            modifier
                .fillMaxWidth()
                .clip(RoundedCornerShape(Radii.lg))
                .background(PantopusColors.appSurface),
    ) {
        rows.forEachIndexed { index, row ->
            Row(
                modifier =
                    Modifier
                        .fillMaxWidth()
                        .padding(Spacing.s3),
                horizontalArrangement = Arrangement.spacedBy(Spacing.s3),
            ) {
                Text(
                    text = row.label,
                    style = PantopusTextStyle.caption,
                    color = PantopusColors.appTextSecondary,
                    modifier = Modifier.width(96.dp),
                )
                Text(
                    text = row.value,
                    style = PantopusTextStyle.body,
                    color = PantopusColors.appText,
                )
            }
            if (index != rows.lastIndex) {
                HorizontalDivider(thickness = 1.dp, color = PantopusColors.appBorderSubtle)
            }
        }
    }
}

File: frontend/apps/ios/Pantopus/Features/Shared/Wizard/Blocks/ReviewSummaryBlock.swift
//
//  ReviewSummaryBlock.swift
//  Pantopus
//
//  Wizard content block — read-only K/V summary card used on review steps.
//

import SwiftUI

/// A single label/value row in a `ReviewSummaryBlock`.
public struct ReviewSummaryRow: Identifiable, Equatable {
    public let id = UUID()
    public let label: String
    public let value: String

    public init(label: String, value: String) {
        self.label = label
        self.value = value
    }
}

/// White card with a stack of label/value rows separated by hairlines.
/// Use on the wizard's review step.
public struct ReviewSummaryBlock: View {
    private let rows: [ReviewSummaryRow]

    public init(_ rows: [ReviewSummaryRow]) {
        self.rows = rows
    }

    public var body: some View {
        VStack(spacing: Spacing.s0) {
            ForEach(Array(rows.enumerated()), id: \.element.id) { index, row in
                HStack(alignment: .firstTextBaseline) {
                    Text(row.label)
                        .pantopusTextStyle(.caption)
                        .foregroundStyle(Theme.Color.appTextSecondary)
                        .frame(width: 96, alignment: .leading)
                    Text(row.value)
                        .pantopusTextStyle(.body)
                        .foregroundStyle(Theme.Color.appText)
                        .frame(maxWidth: .infinity, alignment: .leading)
                }
                .padding(Spacing.s3)
                if index != rows.count - 1 {
                    Rectangle()
                        .fill(Theme.Color.appBorderSubtle)
                        .frame(height: 1)
                }
            }
        }
        .background(Theme.Color.appSurface)
        .clipShape(RoundedRectangle(cornerRadius: Radii.lg, style: .continuous))
        .accessibilityElement(children: .combine)
    }
}
